marketplace-kit-watch.js

```diff
@@ -49,6 +50,11 @@ const isNotEmptyYML = filePath => {
   return true;
 };
 
+// Mdule files outside public or private folders are not synced
+const isModuleFile = f => {
+  return !/^modules\/([^\/]+)\/((?!(public|private)).)*(\/.*)*$/.test(f);
```

Comment:
In general, it's better to not use regexps for this, as they are complicated and error-prone. What you can do is: 
```
'modules/my_cool_module/private/views'.split(path.sep);
// Returns: ['modules', 'my_cool_module', 'private', 'views']
```

By using [path.sep](https://nodejs.org/docs/latest/api/path.html#path_path_sep) you make sure it will work on all OSes. Notice that if you split it into array of directories you only need to check that the 3rd one is either `private` or `public` if the first one is `modules`

Reply:
`my_cool_module` in that case is dynamic value

Reply:
plus I check if it's not public